src/state/actions/__test__/companies.test.js

```diff
@@ -0,0 +1,138 @@
+import * as types from '../../types/index';
+
+import { getCompanies, addCompany } from '../companies';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const mock = new MockAdapter(axios);
+
+const testCompanies = [
+  {
+    id: 1,
+    name: 'Accenture',
+    website: 'www.accenture.com.',
+    location: 'Atlanta, GA',
+    type: 'Business',
+    logo: '',
+    description: '',
+    created_at: null,
+    updated_at: null,
+    latitude: 33.7537,
+    longitude: -85,
+    average_rating: '4.00000000000000000000',
+  },
+];
+
+const testNewCompany = {
+  name: 'Accenture',
+  website: 'www.accenture.com.',
+  location: 'Atlanta, GA',
+  longitude: -85.0,
+  latitude: 33.7537,
+  type: 'Business',
+  logo: '',
+  description: '',
+};
+
+afterEach(rtl.cleanup);
+let wrapper;
+beforeEach(
+  rtl.cleanup
+  //   () => {
+  //   wrapper = rtl.render;
+  // }
+);
+
+describe('Action/types for list of companies(get)', () => {
+  it('should execute get_Companies', async () => {
+    const store = mockStore({});
+    const actions = store.getActions();
+
+    await store.dispatch(getCompanies());
+    expect(actions[0]).toEqual({
+      type: types.GET_COMPANIES,
+    });
+  });
+
+  it('Should return companies array', async () => {
+    await mock
+      .onGet(`${process.env.REACT_APP_BACKEND_URL}/companies/`)
+      .reply(200, testCompanies);
+
+    const expectedActions = {
+      type: types.GET_COMPANIES_SUCCESS,
+      payload: testCompanies,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(getCompanies());
+    expect(actions[1]).toEqual(expectedActions);
+  });
+
+  it('should execute fetch error', async () => {
+    const code = 404;
+    await mock
+      .onGet(`${process.env.REACT_APP_BACKEND_URL}/companies/`)
+      .reply(code);
+    const expectedAction = {
+      type: types.GET_COMPANIES_FAILURE,
+      payload: `Request failed with status code ${code}`,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(getCompanies());
+    expect(actions[1]).toEqual(expectedAction);
+  });
+});
+
+describe('Action/types for adding new company(post)', () => {
+  it('should execute ADD_COMPANY', async () => {
+    const store = mockStore({});
+    const actions = store.getActions();
+
+    await store.dispatch(addCompany());
+    expect(actions[0]).toEqual({
+      type: types.ADD_COMPANY,
+    });
+  });
+
+  it('Should return details of newly added company', async () => {
+    const URL = `${process.env.REACT_APP_BACKEND_URL}/companies/`;
+    const responseObject = {
+      ...testNewCompany,
+      id: 1,
+    };
+    await mock.onPost().reply(201, responseObject);
+
+    const expectedActions = {
+      type: types.ADD_COMPANY_SUCCESS,
+      payload: responseObject,
+    };
+    const store = mockStore({});
+    const actions = store.getActions();
+    await store.dispatch(addCompany());
```

Comment:
Your test passes but it looks to me like you're missing something in your test. This is a test to add a company and it's supposed to take a parameter on the post in order to create a company. Also, on like `119` where you call `store.dispatch(addCompany())`, the `addCompany` action is supposed to take in a parameter which is the test data for the new company to be created. All you seem to be doing is saving the test data in a response object and checking the reply to ensure it returns the response Object. Plus you have a `URL` initialised that you aren't using, and it should definitely be used.

What I would suggest, is mock the `addCompany` properly by sending in a test company data to post then let the reply contain the test data and the id like you have.

Reply:
Good one. I was passing the URL and and test data to "onPost" and tests kept failing. I'd thought the problem was from the mock adapter. Fixed.

Reply:
Great!